joomla/joomla-cms
# [4.0] WebAsset for components, part 3

Third and last part, prev is #29464, #29496, can be tested independently

### Summary of Changes
This make components to use WebAsset,
Affected components:
```
com_finder
    index.php?option=com_finder&view=search
    /administrator/index.php?option=com_finder&view=filter&layout=edit
    /administrator/index.php?option=com_finder&view=filters
    /administrator/index.php?option=com_finder&view=maps
    /administrator/index.php?option=com_finder&view=index
    /administrator/index.php?option=com_finder&view=indexer (indexing popup)
com_modules
    /administrator/index.php?option=com_modules&view=modules&layout=modal
    /administrator/index.php?option=com_modules&view=select&layout=default
    /administrator/index.php?option=com_modules&view=module&layout=edit&id=134
com_menus
    /administrator/index.php?option=com_menus&view=menus
    /administrator/index.php?option=com_menus&view=item&client_id=0&layout=edit
    /administrator/index.php?option=com_menus&view=menutypes (popup in an item edit)
    /administrator/index.php?option=com_menus&view=items
    /administrator/index.php?option=com_menus&view=items&layout=modal
com_joomlaupdate
    /administrator/index.php?option=com_joomlaupdate
    /administrator/index.php?option=com_joomlaupdate&view=update
com_installer
    /administrator/index.php?option=com_installer&view=install
    /administrator/index.php?option=com_installer&view=update
    /administrator/index.php?option=com_installer&view=manage
```


### Testing Instructions
Apply patch, run npm install
Navigate around the site, try visit the component links from the list.
Inspect source code, you should see component scripts loaded.


### Expected result
All works


### Actual result
All works


ref #22435
